ZetaPushSwift/ClientHelper.swift

```diff
@@ -1,366 +1,349 @@
+    cometdClient.publish(message, channel: channel)
+  }
+  
+  open func unsubscribe(_ subscription:Subscription){
+    log.zp.debug("ClientHelper unsubscribe")
+    cometdClient.unsubscribeFromChannel(subscription)
+    if let index = subscriptionQueue.index(of: subscription){
+      subscriptionQueue.remove(at: index)
+    }
+  }
+  
+  open func logout(){
+    log.zp.debug("ClientHelper logout")
+    eraseHandshakeToken()
+    disconnect()
+  }
+  
+  open func setForceSecure(_ isSecure: Bool){
+    cometdClient.setForceSecure(isSecure)
+  }
+  
+  open func composeServiceChannel(_ verb: String, deploymentId: String) -> String {
+    return "/service/" + sandboxId + "/" + deploymentId + "/" + verb
+  }
+  
+  open func getLogLevel() -> XCGLogger.Level {
+    return logLevel
+  }
+  
+  open func setLogLevel(logLevel: XCGLogger.Level){
+    self.logLevel = logLevel
+    log.setup(level: logLevel)
+  }
+  
+  /*
+   Must be overriden by ClientHelper descendants
+   */
+  func storeHandshakeToken(_ authenticationDict: NSDictionary) { }
+  /*
+   Must be overriden by ClientHelper descendants
+   */
+  func eraseHandshakeToken() { }
+  
+  open func getClientId() -> String {
+    return cometdClient.getCometdClientId()
+  }
+  
+  open func getHandshakeFields() -> [String: Any] {
+    return authentication!.getHandshakeFields(self)
+  }
+  
+  open func getResource() -> String {
+    return resource
+  }
+  
+  open func getSandboxId() -> String {
+    return sandboxId
+  }
+  
+  open func getServer() -> String {
+    return server
+  }
+  
+  open func setServerUrl(_ serverUrl: String) {
+    server = serverUrl
+  }
+  
+  open func getUserId() -> String {
+    return userId
+  }
+  
+  open func isConnected() -> Bool {
+    return cometdClient.isConnected()
+  }
+  
+  open func getPublicToken() -> String {
+    return publicToken
+  }
+  
+  open func isWeaklyAuthenticated() -> Bool {
+    return !publicToken.isEmpty
+  }
+  
+  open func isStronglyAuthenticated() -> Bool {
+    return !isWeaklyAuthenticated() && !token.isEmpty
+  }
+  
+  /*
+   Delegate functions from CometdClientDelegate
+   */
+  open func connectedToServer(_ client: CometdClient) {
+    log.zp.debug("ClientHelper Connected to ZetaPush server")
+    wasConnected = connected
+    connected = true
+    if !wasConnected && connected {
+      delegate?.onConnectionEstablished(self)
+    }
+  }
+  
+  open func handshakeSucceeded(_ client: CometdClient, handshakeDict: NSDictionary) {
+    log.zp.debug("ClientHelper Handshake Succeeded")
+    log.zp.debug(handshakeDict)
+    let authentication: NSDictionary = handshakeDict["authentication"] as? NSDictionary ?? [:]
+    
+    if authentication["token"] != nil {
+      token = (authentication["token"] as? String) ?? ""
+    }
+    
+    if authentication["publicToken"] != nil {
+      publicToken = authentication["publicToken"] as? String ?? ""
+    }
+    
+    userId = authentication["userId"] as? String ?? ""
```

Comment:
here you have two options, use a guard and log if an error occurs, or use `if let` like before